src/gui/providers/qgspointcloudsourceselect.cpp

```diff
@@ -64,6 +78,58 @@ QgsPointCloudSourceSelect::QgsPointCloudSourceSelect( QWidget *parent, Qt::Windo
 
 void QgsPointCloudSourceSelect::addButtonClicked()
 {
+  QStringList openOptions;
+  for ( QWidget *control : mOpenOptionsWidgets )
+  {
+    QString value;
+    if ( control->objectName() == QStringLiteral( "separator" ) )
+    {
+      const QComboBox *comboBox = control->findChild<QComboBox *>();
+      if ( comboBox )
+      {
+        const int currentData = comboBox->itemData( comboBox->currentIndex() ).toInt();
+        int valueAsChar = 32; // space is the default separator
+        if ( currentData == QgsPointCloudSourceSelect::SEPARATOR_OTHER )
+        {
+          const QLineEdit *otherWidget = control->findChild<QLineEdit *>();
```

Comment:
This approach is fragile. I'd suggest creating a private widget class for the separator widget and then just accessing it directly here via qobject_casting control to the private widget class.